api.go

```diff
@@ -27,10 +27,10 @@ type API struct {
 	FilterStructMemberGetter func(m *StructMember) bool
 }
 
-func (a *API) HandleDirectory(dir string) error {
+func (a *API) HandleDirectory(dir string) (*HeaderFile, error) {
```

Comment:
Yeah, HeaderFile is indeed not really correctly named, I will plan out the "new" PR today.